tools/release-post-generator/commands/release-post.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * External dependencies
+ */
+import Analyzer from 'code-analyzer/src/commands/analyzer';
+import semver from 'semver';
+import { promises } from 'fs';
+import { writeFile } from 'fs/promises';
+
+/**
+ * Internal dependencies
+ */
+import { program } from '../program';
+import { renderTemplate } from '../lib/render-template';
+import { processChanges } from '../lib/process-changes';
+import { createWpComDraftPost } from '../lib/draft-post';
+import { generateContributors } from '../lib/contributors';
+import { Logger } from '../lib/logger';
+
+const VERSION_VALIDATION_REGEX =
+	/^([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$/;
+
+// Define the release post command
+program
+	.command( 'release' )
+	.description( 'CLI to automate generation of a release post.' )
+	.argument(
+		'<currentVersion>',
+		'The version of the plugin to generate a post for, please use the tag version from Github.'
+	)
+	.option( '--outputOnly', 'Only output the post, do not publish it' )
+	.option(
+		'--previousVersion <previousVersion>',
+		'If you would like to compare against a version other than last minor you can provide a tag version from Github.'
+	)
+	.action( async ( currentVersion, options ) => {
+		const previousVersion = options.previousVersion
+			? semver.parse( options.previousVersion )
```

Comment:
parsing versions with node-semver is convenient for simple version manipulation and not having to rely on regex.